go/libraries/doltcore/merge/merge_prolly_rows.go

```diff
@@ -1266,12 +1272,56 @@ func remapTupleWithColumnDefaults(ctx *sql.Context, keyTuple, valueTuple val.Tup
 				}
 			}
 		} else {
-			tb.PutRaw(to, tupleDesc.GetField(from, valueTuple))
+			value, err := index.GetField(ctx, tupleDesc, from, valueTuple, tm.ns)
+			if err != nil {
+				return nil, err
+			}
+
+			// If the type has changed, then call convert to convert the value to the new type
+			value, err = convertValueToNewType(value, col.TypeInfo, tm, from, rightSide)
+			if err != nil {
+				return nil, err
+			}
+
+			err = index.PutField(ctx, tm.ns, tb, to, value)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
 	return tb.Build(pool), nil
 }
 
+// convertValueToNewType handles converting a value from a previous type into a new type. |value| is the value from
+// the previous schema, |newTypeInfo| is the type info for the value in the new schema, |tm| is the TableMerger
+// instance that describes how the table is being merged, |from| is the field position in the value tuple from the
+// previous schema, and |rightSide| indicates whether the previous type info can be found on the right side of the merge
+// or the left side. If the previous type info is the same as the current type info for the merged schema, then this
+// function is a no-op and simply returns |value|. The converted value along with any unexpected error encountered is
+// returned.
+func convertValueToNewType(value interface{}, newTypeInfo typeinfo.TypeInfo, tm *TableMerger, from int, rightSide bool) (interface{}, error) {
+	var previousTypeInfo typeinfo.TypeInfo
+	if rightSide {
+		previousTypeInfo = tm.rightSch.GetNonPKCols().GetByIndex(from).TypeInfo
+	} else {
+		previousTypeInfo = tm.leftSch.GetNonPKCols().GetByIndex(from).TypeInfo
+	}
+
+	// If the type has changed, then call convert to convert the value to the new type
+	if newTypeInfo.Equals(previousTypeInfo) {
+		return value, nil
+	}
+
+	newValue, inRange, err := newTypeInfo.ToSqlType().Convert(value)
+	if err != nil {
+		return nil, err
+	}
+	if !inRange {
+		return nil, fmt.Errorf("out of range conversion for value %v to type %s", value, newTypeInfo.String())
```

Comment:
I'll think about this one a little more... I don't see a great way to trigger this path currently since we shouldn't ever execute this code unless we've already verified that the type we're moving to is larger than the type we're coming from and `StringType.Convert` will only return out of range when we give it a value larger than the type's width.